wagtail/admin/forms/workflows.py

```diff
@@ -1,9 +1,77 @@
 from django import forms
 from django.core.exceptions import ValidationError
-from django.utils.translation import ugettext_lazy as _
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext as _
+from django.utils.translation import ugettext_lazy as __
 
 from wagtail.admin import widgets
-from wagtail.core.models import Page, Workflow, WorkflowPage
+from wagtail.core.models import Page, Task, Workflow, WorkflowPage
+
+
+def model_name(model):
+    return model._meta.app_label + '.' + model.__name__
+
+
+class TaskChooserSearchForm(forms.Form):
+    q = forms.CharField(label=__("Search term"), widget=forms.TextInput(), required=False)
+
+    def __init__(self, *args, task_type_choices=None, **kwargs):
+        placeholder = kwargs.pop('placeholder', _("Search"))
+        super().__init__(*args, **kwargs)
+        self.fields['q'].widget.attrs = {'placeholder': placeholder}
+
+        # Add task type filter if there is more than one task type option
+        if task_type_choices and len(task_type_choices) > 1:
+            self.fields['task_type'] = forms.ChoiceField(
+                choices=(
+                    # Append an "All types" choice to the beginning
+                    [(None, _("All types"))]
+
+                    # The task type choices that are passed in use the models as values, we need
+                    # to convert these to something that can be represented in HTML
+                    + [
+                        (model_name(model), verbose_name)
+                        for model, verbose_name in task_type_choices
+                    ]
+                ),
+                required=False
+            )
+
+        # Save a mapping of task_type values back to the model that we can reference later
+        self.task_type_choices = {
+            model_name(model): model
+            for model, _ in task_type_choices
```

Comment:
can you rework this in light of @gasman's comment - https://github.com/wagtail/wagtail/pull/6163#issuecomment-647579099:

> for example, using _ as the return value of get_or_create is liable to clash with the _ alias commonly used for gettext